src/extractor/guidance/intersection_generator.cpp

```diff
@@ -55,8 +55,26 @@ Intersection IntersectionGenerator::GetConnectedRoads(const NodeID from_node,
 {
     Intersection intersection;
     const NodeID turn_node = node_based_graph.GetTarget(via_eid);
-    const NodeID only_restriction_to_node =
-        restriction_map.CheckForEmanatingIsOnlyTurn(from_node, turn_node);
+    const NodeID only_restriction_to_node = [&]() {
+        // If only restrictions refer to invalid ways somewhere far away, we rather ignore the
+        // restriction than to not route over the intersection at all.
+        const auto only_restriction_to_node =
+            restriction_map.CheckForEmanatingIsOnlyTurn(from_node, turn_node);
+        if (only_restriction_to_node != SPECIAL_NODEID)
+        {
+            for (const EdgeID onto_edge : node_based_graph.GetAdjacentEdgeRange(turn_node))
+            {
+                BOOST_ASSERT(onto_edge != SPECIAL_EDGEID);
+                const NodeID to_node = node_based_graph.GetTarget(onto_edge);
+                if (to_node == only_restriction_to_node)
+                {
+                    return only_restriction_to_node;
+                }
+            }
+            // Ignore broken only restrictions.
```

Comment:
Good catch :+1: looks good.

There are range algorithms for this pattern, too, as in

    boost::find_if(EdgeRange, [] (EdgeId e) { .. });